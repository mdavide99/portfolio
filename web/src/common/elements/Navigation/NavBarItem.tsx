import Link from "next/link";
import { useRouter } from "next/router";
import classNames from "classnames";

interface NavBarItemProps {
  name: string;
  href: string;
  className?: string;
}

function NavBarItem(props: NavBarItemProps) {
  const { asPath } = useRouter();
  const isActive =
    props.href === "/" ? asPath === "/" : asPath.startsWith(props.href);
  let linkClass = classNames(
    "text-4xl sm:text-base my-3 sm:my-0 mx-5 text-gray-500 dark:text-neutral-100 hover:text-gray-900 hover:dark:text-white hover:transition-color hover:underline decoration-blue-500 dark:decoration-fuchsia-500 decoration-4 sm:decoration-2 underline-offset-8",
    { "text-gray-900 underline": isActive }
  );

  return (
    <Link href={props.href}>
      <a className={linkClass}>{props.name}</a>
    </Link>
  );
}

export { NavBarItem };
