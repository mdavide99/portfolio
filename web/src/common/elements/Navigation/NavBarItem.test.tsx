import { NavBarItem } from "./NavBarItem";
import { render } from "@testing-library/react";
jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn().mockReturnValue({
    route: "",
    pathname: "",
    query: "",
    asPath: "",
  }),
}));

jest.mock("next/link", () => {
  const React = require("react");
  return ({ children, href }: React.PropsWithChildren<{ href: string }>) =>
    React.cloneElement(React.Children.only(children), { href });
});
describe("NavBarItem", () => {
  it("renders NavBarItem correctly", () => {
    const { container } = render(<NavBarItem name={"TEST"} href={"#"} />);
    expect(container).toMatchSnapshot();
  });
});
