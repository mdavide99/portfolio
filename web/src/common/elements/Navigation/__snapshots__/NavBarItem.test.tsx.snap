// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarItem renders NavBarItem correctly 1`] = `
<div>
  <a
    class="text-4xl sm:text-base my-3 sm:my-0 mx-5 text-gray-500 dark:text-neutral-100 hover:text-gray-900 hover:dark:text-white hover:transition-color hover:underline decoration-blue-500 dark:decoration-fuchsia-500 decoration-4 sm:decoration-2 underline-offset-8"
    href="#"
  >
    TEST
  </a>
</div>
`;
